packages/perf-react/src/lexical/plugins/ContextMenuPlugin.tsx

```diff
@@ -0,0 +1,189 @@
+  }
+  return (
+    <li
+      key={option.key}
+      tabIndex={-1}
+      className={className}
+      ref={option.setRefElement}
+      role="option"
+      aria-selected={isSelected}
+      id={"typeahead-item-" + index}
+      onMouseEnter={onMouseEnter}
+      onClick={onClick}
+    >
+      <span className="text">{option.title}</span>
+    </li>
+  );
+}
+
+function ContextMenu({
+  options,
+  selectedItemIndex,
+  onOptionClick,
+  onOptionMouseEnter,
+}: {
+  selectedItemIndex: number | null;
+  onOptionClick: (option: ContextMenuOption, index: number) => void;
+  onOptionMouseEnter: (index: number) => void;
+  options: Array<ContextMenuOption>;
+}) {
+  return (
+    <div className="typeahead-popover">
+      <ul>
+        {options.map((option: ContextMenuOption, i: number) => (
+          <ContextMenuItem
+            index={i}
+            isSelected={selectedItemIndex === i}
+            onClick={() => onOptionClick(option, i)}
+            onMouseEnter={() => onOptionMouseEnter(i)}
+            key={option.key}
+            option={option}
+          />
+        ))}
+      </ul>
+    </div>
+  );
+}
+
+export class ContextMenuOption extends MenuOption {
+  title: string;
+  onSelect: (targetNode: LexicalNode | null) => void;
+  constructor(
+    title: string,
+    options: {
+      onSelect: (targetNode: LexicalNode | null) => void;
+    },
+  ) {
+    super(title);
+    this.title = title;
+    this.onSelect = options.onSelect.bind(this);
+  }
+}
+
+export default function ContextMenuPlugin(): JSX.Element {
+  const [editor] = useLexicalComposerContext();
+
+  const options = useMemo(() => {
+    return [
+      new ContextMenuOption(`Copy`, {
+        onSelect: (_node) => {
+          editor.dispatchCommand(COPY_COMMAND, null);
+        },
+      }),
+      new ContextMenuOption(`Cut`, {
+        onSelect: (_node) => {
+          editor.dispatchCommand(CUT_COMMAND, null);
+        },
+      }),
+      new ContextMenuOption(`Paste`, {
+        onSelect: (_node) => {
+          navigator.clipboard.read().then(async () => {
+            const data = new DataTransfer();
+
+            const items = await navigator.clipboard.read();
+            const item = items[0];
+
+            const permission = await navigator.permissions.query({
+              // @ts-expect-error These types are incorrect.
+              name: "clipboard-read",
+            });
+            if (permission.state === "denied") {
+              alert("Not allowed to paste from clipboard.");
+              return;
+            }
+
+            for (const type of item.types) {
+              const dataString = await (await item.getType(type)).text();
+              data.setData(type, dataString);
+            }
+
+            const event = new ClipboardEvent("paste", {
+              clipboardData: data,
+            });
+
+            editor.dispatchCommand(PASTE_COMMAND, event);
+          });
+        },
+      }),
+    ];
+  }, [editor]);
```

Comment:
Is there some reason you didn't include the **Paste as Plain Text** option? If it works properly then please add it in. If it caused some issue then leave it out.